<launch>
<arg name="robot"/>
<!-- autostart: true to start the PPLP at startup -->
<arg name="autostart" default="false"/>
<!-- display: true to show images -->
<arg name="display" default="false"/>
<!-- eventgui: true to create a GUI that allows to start the PPLP -->
<arg name="eventgui"  default="false"/>
<!-- marker: true to create a node converting the published PPL into a Vizualisation marker -->
<arg name="marker"    default="false"/>

<group ns="$(arg robot)">
  <!-- HOG people detector
      ROS_NAMESPACE=alz rosrun people_detection_vision hog_pplp.exe -activate -display _data_acquisition_by_wait_for_message:=false
    -->
  <node pkg="people_detection_vision" type="hog_pplp.exe" name="hog_pplp" respawn="true"
        output="screen" args="_resize_scale:=1
                              _hog_hitThreshold:=0
                              _data_acquisition_want_depth:=true
                              ">
    <!-- autostart param -->
    <param name="autostart" value="$(arg autostart)"/>
    <!-- display param -->
    <param name="display" value="$(arg display)"/>
  </node>
<!--
                              _data_acquisition_by_wait_for_message:=true
                              _resize_scale:=0.4
                              _hog_hitThreshold:=2.0
-->

  <!-- conversion to ROS rviz marker -->
  <node if="$(arg marker)" pkg="vision_utils" type="ppl2marker.exe"
        output="screen" name="ppl2marker_people_hog_pplp"
        args="_red:=0 _green:=0 _blue:=1 _input_topic:=hog_pplp/ppl" />

  <!-- event GUI -->
  <node if="$(arg eventgui)" pkg="ad_core" type="launcher_qt_event_sender_interface.exe"
        name="hog_event_gui"
        output="screen"
        args='event_keys="|HOG_PPLP_START;HOG_PPLP_START;0
                          |HOG_PPLP_STOP;HOG_PPLP_STOP;0"'/>
</group>
</launch>

